doc/explanation/scopes.rst

```diff
@@ -0,0 +1,39 @@
+Dune Projects and Workspaces
+============================
+
+Whenever Dune builds anything, it does so at the level of a *Dune workspace*. A
+Dune workspace is a set of Dune projects. A typical workspace consists of a
+single Dune project.
+
+A *Dune project* is defined by the presence of a
+:doc:`/reference/dune-project/index` file. Each Dune project extends over the
+file tree rooted at the directory containing the
+:doc:`/reference/dune-project/index` file, excluding any nested Dune projects.
+
+Dune determines the root of the current workspace by finding the topmost
+ancestor containing a :doc:`/reference/dune-project/index` file or by the
+presence of a :doc:`/reference/dune-workspace/index` file (see
+:ref:`finding-root` and :ref:`forcing-root` for details).
+
+Different Dune projects within the same Dune workspace are independent of each
+other and no settings are shared between them, even if they are nested within
+each other.
+
+Settings in :doc:`/reference/dune-workspace/index`, on the other hand, are
+inherited by all Dune projects in the workspace. Some settings (those that make
+sense for all projects) can be specified both in
+:doc:`/reference/dune-project/index` and :doc:`/reference/dune-workspace/index`
+files, with the former taking precedence. Note that all
+:doc:`/reference/dune-workspace/index` files other than the one specifying the
+root of the workspace are ignored.
+
+Within a Dune project, :doc:`/reference/dune/index` files are used to define all
+objects of interest for Dune: libraries, executables, tests, etc. References
+within :doc:`/reference/dune/index` files are resolved relative to the directory
+containing the file (except if using a :doc:`/reference/dune/subdir` stanza or
+the :doc:`/reference/actions/chdir` action`). There are typically many
+:doc:`/reference/dune/index` files in a Dune project (one per directory, unless
+the directory does not contain anything relevant to Dune).
```

Comment:
```suggestion
:doc:`/reference/dune/index` files in a Dune project (one per directory, unless
the directory does not contain anything relevant to Dune or the sub directory
is a descendant of a directory using :doc:`/reference/dune/include_subdirs`).
```
Should we add a reference to `(include_subdirs [...])`? I'm asking because this is a useful and underrated feature and it has its place in this part of the documentation. (I'm not sure if the suggestion is correctly working...)

Reply:
Actually, I would rather keep in this section only the explanations of the general case, and indicate that these general rules have exceptoins, pointing the reader to the specific stanzas (`(subdir)`, `(include_subdirs)`, `(chdir)`, etc) to learn about the details.

I will push a commit in this direction shortly.

Reply:
Addressed in 6602c2a02abf2737f51e08f031e9b167ca6c827b